src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -103,7 +103,7 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
         Address updatedAddress = editPersonDescriptor.getAddress().orElse(personToEdit.getAddress());
         Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToEdit.getTags());
-        List<Task> tasks = personToEdit.getTasks();
+        List<Task> tasks = editPersonDescriptor.getTasks().orElse(personToEdit.getTasks());
```

Comment:
Thanks for pointing that out. 
Actually, EditCommand overwrites the entire taskList, so the function of EditCommand and EditTaskCommand is distinct in terms of manipulating tasks.
I also made the changes because due to regression from the observable task list implementation, some test could not pass and that was the only way to fix the issue.